source/Octopus.Tentacle.Tests/Commands/MigratePreInstalledKubernetesDeploymentTargetCommandTest.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+using FluentAssertions;
+using k8s.Models;
+using NSubstitute;
+using NUnit.Framework;
+using Octopus.Tentacle.Commands;
+
+namespace Octopus.Tentacle.Tests.Commands
+{
+    [TestFixture]
+    public class MigratePreInstalledKubernetesDeploymentTargetCommandFixture
+    { 
+        [Test]
+        public void ShouldNotMigrate_IfNoSource()
+        {
+            var targetConfigMap = Substitute.For<V1ConfigMap>();
+            var targetSecret = Substitute.For<V1Secret>();
+            
+            MigratePreInstalledKubernetesDeploymentTargetCommand.ShouldMigrateData(null, null, targetConfigMap, targetSecret).Item1.Should().BeFalse();
```

Comment:
Now that this method returns named values, you can replace `Item1` with `ShouldMigrate` here and for the other tests.

On reading this with fresh eyes I'd actually suggest split the act and assert for clarity
```suggestion
            var (shouldMigrate, _) = MigratePreInstalledKubernetesDeploymentTargetCommand.ShouldMigrateData(null, null, targetConfigMap, targetSecret);
            
            shouldMigrate.Should().BeFalse()
```

Reply:
And to be honest, it's probably valid to just check that the correct error message is being shown here too

Reply:
Done!